### PR TITLE
[matterircd_complete.pl] Add new matterircd_complete.pl script

### PR DESCRIPTION
**Pull request by hloeung**
For https://github.com/42wim/matterircd networks, supports tab auto completion (or key binding)
for nicks and message/threads.

### CODE REVIEW

**Comment by ailin-nemui** [comment]
sounds interesting!

**Comment by hloeung** [comment]
> sounds interesting!

Thanks. I think useful for those using [matterircd](https://github.com/42wim/matterircd) - a [Mattermost](https://mattermost.org/) (and Slack) to IRC bridge.

File: scripts/matterircd_complete.pl [hunk 1]
```diff
@@ -0,0 +1,815 @@
...
+        gui_input_set_pos($pos);
+
+        $MSGTHREADID_CACHE_INDEX = 0;
+        $MSGTHREADID_CACHE_SEARCH_ENABLED = 0;
+
+        $ESC_PRESSED = 0;
+        $O_PRESSED = 0;
+    }
+
+    # For 'down arrow', it's a sequence of ESC + O + B.
```
Comment (ailin-nemui) [inline scripts/matterircd_complete.pl:297]: note that these key sequences may differ between terminal emulators.

File: scripts/matterircd_complete.pl [hunk 2]
```diff
@@ -0,0 +1,815 @@
...
+#   /set matterircd_complete_message_thread_id_cache_size 50
+#   /set matterircd_complete_nick_cache_size 20
+#
+# To ignore specific nicks in autocomplete:
+#
+#   /set matterircd_complete_nick_ignore somebot anotherbot
+
+use strict;
+use warnings;
+use experimental 'smartmatch';
```
Comment (ailin-nemui) [inline scripts/matterircd_complete.pl:57]: if you want, you could `use List::Util 'any';` and `any { $_ eq $needle } @array ` instead of `~~`

**Comment by ailin-nemui** [comment]
@hloeung are you satisfied with the script and should it be added to the repo?

**Comment by hloeung** [comment]
@ailin-nemui , yes satisfied with the script as is and for it to be added to the repo. Thanks!

**Pull request merged**